pkg/virt-api/webhooks/defaults.go

```diff
@@ -69,7 +69,11 @@ func setupHotplug(clusterConfig *virtconfig.ClusterConfig, vmi *v1.VirtualMachin
 
 func setupCPUHotplug(clusterConfig *virtconfig.ClusterConfig, vmi *v1.VirtualMachineInstance) {
 	if vmi.Spec.Domain.CPU.MaxSockets == 0 {
-		vmi.Spec.Domain.CPU.MaxSockets = clusterConfig.GetMaximumCpuSockets()
+		maxSockets := clusterConfig.GetMaximumCpuSockets()
+		if vmi.Spec.Domain.CPU.Sockets > maxSockets && maxSockets != 0 {
```

Comment:
nit: why do you need to check that `maxSockets != 0` ?

Reply:
I'm not a huge fan of helpers like `GetMaximumCpuSockets` hiding that `MaxCpuSockets` is actually a pointer. It doesn't massively matter in this case but it's much easier to assert if something isn't configured by checking for `nil` as opposed to default values like `0` or `false` with bools.